security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -968,6 +971,13 @@ func (sc *SecretCache) sendRetriableRequest(ctx context.Context, csrPEM []byte,
 		var httpRespCode int
 		if isCSR {
 			requestErrorString = fmt.Sprintf("%s CSR", logPrefix)
+			// if cert exists in the file path user provides,
+			// we will use cert instead of the token to do CSR Sign request
+
+			if err = sc.certExists(); err == nil {
```

Comment:
How about if sc.certExists() {...}, let certExists() return boolean, since the caller does not care what err it is.

Reply:
yes modify the code